dll/win32/kernel32/client/proc.c

```diff
@@ -4377,6 +4486,176 @@ CreateProcessInternalW(IN HANDLE hUserToken,
+												    	DPRINT("CreateProcessInternalW: WSADuplicateSocketW returned %x\n", result);
+													    DPRINT("CreateProcessInternalW: handlevalue used as socket is: %lx\n", handleInfo.HandleValue);
+    													DPRINT("CreateProcessInternalW: CatalogId returned is %x\n", SharedSockets[sockindex].dwCatalogEntryId );
+	    												DPRINT("CreateProcessInternalW: SocketType returned is %x\n", SharedSockets[sockindex].iSocketType );
+		    											DPRINT("CreateProcessInternalW: Protocol returned is %x\n", SharedSockets[sockindex].iProtocol );
+			    										DPRINT("CreateProcessInternalW: dwProviderReserved returned is %x\n", SharedSockets[sockindex].dwProviderReserved );
+				    									// DuplicateSocket populates these for us, we can use dwProviderReserved item to carry handle value and be our flag that socket is inherited.
+					    								SharedSockets[sockindex].dwProviderReserved = handleInfo.HandleValue;
+						    							DPRINT("CreateProcessInternalW: dwProviderReserved overriden to %x\n", SharedSockets[sockindex].dwProviderReserved );
+							    						if (result == 0) {
+								    						sockindex = sockindex + 1; // increment array index and counter
+									    					DPRINT("CreateProcessInternalW: WSADuplicateSocketW successful, duplicated (%x) sockets so far\n", sockindex);
+										    			} else DPRINT1("CreateProcessInternalW: call to WSADuplicateSocketW failed, returned:%x\n", result);
+											    	}
+    											} else DPRINT1("Error calling getsockopt for inheritable socket\n");
+	    									} else DPRINT1("Error dynlinking getsockopt function (GetProcAddressA returned NULL)\n");
+		    							} else DPRINT1("Error getting module handle for ws2_32 DLL(GetModuleHandleA returned NULL)\n");
+			    					} 
+				    			} 
+					    	} 
+    					} 
+	    			} 
+		    	} //end of for loop scanning system handle table
+			    CloseHandle(PROCESS_HANDLE);
+    		} else { DPRINT1("CreateProcessInternalW: winsock Handle scan failed to open handle to own process\n"); } 
+	    } else { DPRINT1("CreateProcessInternalW: system Handle Table scan routine failed to reference system Handle table\n"); } 
+	    DPRINT("CreateProcessInternalW: System Handle Table scan routine completed\n");			
+
+    	if (sockindex > 0) 
+	    {
+		    DPRINT1("CreateProcessInternalW: creating winsock memory structures in child process and marking PEB\n");			
+		    Size = sockindex * sizeof( WSAPROTOCOL_INFOW );
+    		DPRINT("CreateProcessInternalW: calling NtAllocateVirtualMemory() with socket block size:%lx\n", Size);
+	    	RemoteWinsockData = NULL; // init just in case its not null
+		    Status = NtAllocateVirtualMemory(ProcessHandle,
+                        						&RemoteWinsockData,
+                        						0,
+                        						&Size,
+                        						MEM_COMMIT,
+						                        PAGE_READWRITE);
```

Comment:
Again the indentation level is wrong for the lines above. Again you indented partially with tabs here.

Reply:
cleaned up in notepad++